include/dali/core/span.h

```diff
@@ -58,10 +59,30 @@ class span {
   >>
   DALI_HOST_DEV constexpr span(const span<U, Extent>& s) noexcept : data_(s.data()) {}
 
+
   DALI_HOST_DEV constexpr span(pointer firstElem, pointer lastElem) : data_(firstElem) {
     /* assert(lastElem - firstElem == Extent); */
   }
 
+  /**
+   * @remarks Container is not a specialization of span,
+   *          specialization of std::array or an array type
+   */
+  template <typename Container,
+          typename std::enable_if<is_container<Container>::value, int>::type = 0>
```

Comment:
`is_container` is not a good type trait here. We want something that has two methods:
`data()`, returning a pointer
`size()` returning an integer
We also want to reject `std::array` and `span`, since they have specialized constructors.

Moreover, I added the r-value check to `make_span` on purpose - to disallow creating a span from a temporary container.